modules/collect/src/main/java/com/opengamma/strata/collect/result/ValueWithFailures.java

```diff
@@ -87,6 +89,27 @@
   }
 
   //-------------------------------------------------------------------------
+
+  /**
+   * Returns a {@link Collector} which performs a reduction of its
+   * {@link ValueWithFailures} input elements under a specified {@link BinaryOperator} using the
+   * provided identity.
+   *
+   * @param identityValue the identity value
+   * @param operator the operator used for the reduction.
+   * @param <T> the type of the success value in the {@link ValueWithFailures}
+   * @return a {@link Collector}
+   */
+  public static <T> Collector<ValueWithFailures<T>, ?, ValueWithFailures<T>> toValueWithFailures(
+      T identityValue,
+      BinaryOperator<T> operator) {
+
+    BinaryOperator<ValueWithFailures<T>> reduceFunction =
+        (result1, result2) -> result1.combinedWith(result2, operator);
+
+    return Collectors.reducing(ValueWithFailures.of(identityValue), reduceFunction);
+  }
+
```

Comment:
Should be a separator after static methods and before instance ones (because there is already in this class)